src/main/scala/com/microsoft/hyperspace/index/zordercovering/ZOrderField.scala

```diff
@@ -0,0 +1,569 @@
+      }
+      val valueTime = util.DateTimeUtils.fromJavaTimestamp(value)
+      longField.getBitSet(valueTime)
+    }
+  }
+}
+
+/**
+ * For Date type, handle it as Int type after converting it as the number of days.
+ * See [[IntZOrderField]].
+ */
+case class DateMinMaxZOrderField(
+    override val name: String,
+    minVal: java.sql.Date,
+    maxVal: java.sql.Date)
+    extends ZOrderField {
+
+  val intField =
+    IntMinMaxZOrderField(
+      name,
+      util.DateTimeUtils.fromJavaDate(minVal),
+      util.DateTimeUtils.fromJavaDate(maxVal))
+
+  override val bitLen: Int = {
+    intField.bitLen
+  }
+
+  override def getBitSet(valueAny: Any): BitSet = {
+    if (checkEmptySetEligible(valueAny)) {
+      emptyBitSet
+    } else {
+      val value = valueAny.asInstanceOf[java.sql.Date]
+      if (value.compareTo(minVal) < 0 || value.compareTo(maxVal) > 0) {
+        throwOutOfRangeException(minVal, maxVal, value)
+      }
+      val valueDay = util.DateTimeUtils.fromJavaDate(value)
+      intField.getBitSet(valueDay)
+    }
+  }
+}
+
+object ZOrderField {
+  def build(
+      name: String,
+      dataType: DataType,
+      minVal: Any,
+      maxVal: Any,
+      quantiles: Seq[Any] = Nil,
+      quantileEnabled: Boolean = false): ZOrderField = {
+    dataType match {
+      case LongType =>
+        if (quantileEnabled) {
+          LongPercentileZOrderField(
+            name,
+            (minVal +: quantiles :+ maxVal).asInstanceOf[Seq[Double]])
+        } else {
+          LongMinMaxZOrderField(name, minVal.asInstanceOf[Long], maxVal.asInstanceOf[Long])
+        }
+      case IntegerType =>
+        if (quantileEnabled) {
+          IntPercentileZOrderField(
+            name,
+            (minVal +: quantiles :+ maxVal).asInstanceOf[Seq[Double]])
+        } else {
+          IntMinMaxZOrderField(name, minVal.asInstanceOf[Int], maxVal.asInstanceOf[Int])
+        }
+      case ShortType =>
+        if (quantileEnabled) {
+          ShortPercentileZOrderField(
+            name,
+            (minVal +: quantiles :+ maxVal).asInstanceOf[Seq[Double]])
+        } else {
+          ShortMinMaxZOrderField(name, minVal.asInstanceOf[Short], maxVal.asInstanceOf[Short])
+        }
+      case ByteType =>
+        if (quantileEnabled) {
+          BytePercentileZOrderField(
+            name,
+            (minVal +: quantiles :+ maxVal).asInstanceOf[Seq[Double]])
+        } else {
+          ByteMinMaxZOrderField(name, minVal.asInstanceOf[Byte], maxVal.asInstanceOf[Byte])
+        }
+      case DoubleType =>
+        if (quantileEnabled) {
+          DoublePercentileZOrderField(
+            name,
+            (minVal +: quantiles :+ maxVal).asInstanceOf[Seq[Double]])
+        } else {
+          DoublePercentileZOrderField(
+            name,
+            Seq(minVal.asInstanceOf[Double], maxVal.asInstanceOf[Double]))
+        }
+      case FloatType =>
+        // minVal, maxVal can be Float.
```

Comment:
I think the comment should be the opposite to make sense? When can minVal/maxVal can be `Double`?

Reply:
If it's from `approxQuantile` then the value type is Double. If it's from aggs(min, max), then it's Float type.